test/integration/listener_lds_integration_test.cc

```diff
@@ -1598,14 +1598,15 @@ TEST_P(ListenerFilterIntegrationTest,
       {
         sendLdsResponse({MessageUtil::getYamlStringFromMessage(listener_config_)}, "2");
         test_server_->waitForCounterGe("listener_manager.lds.update_rejected", 1);
-        IntegrationTcpClientPtr tcp_client2 = makeTcpConnection(lookupPort("test_listener"));
```

Comment:
Not sure I understand - the macro `EXPECT_LOG_CONTAINS` executes `setShouldEscape` and the code block sequentially. Where's the data race?

Reply:
emm...indeed. 

The log said the write race is in the EXPECT_LOG_CONTANTS, but indeed, that happened before this code bock.

I can't reproduce this local, but I'm guessing this log 
https://dev.azure.com/cncf/envoy/_build/results?buildId=130696&view=logs&j=d1f76054-8f79-554b-6f4a-11d6a63b8e00&t=266e17e3-d213-54b5-deef-0dcee01da137&l=767

may happened before a remote close
https://github.com/envoyproxy/envoy/blob/fa5f548d05da6495c175b5c124292fa0bae215de/source/common/network/raw_buffer_socket.cc#L24-L29

so the real race is at line 1584 closed the client client.
https://github.com/envoyproxy/envoy/pull/26098/files#diff-7fe1ad3a088b5d0ea44d9bace0dfa0675d5009bcb538afb778ebad50c802ddeaR1584

That trigger a remote close for upstream connection. And at same time EXPECT_LOG_CONTANTS executes the `setShouldEscape`

Let me try to update this.

Thanks for the review again!